src/components/features/NewDeliveryTable.tsx

```diff
@@ -0,0 +1,215 @@
+
+const markers = ['Масло', 'Гуми', 'Чистачки']
+
+export default function NewDeliveryTable() {
+  const { t: translate } = useTranslation()
+  const [searchTerm, setSearchTerm] = React.useState('')
+  const [deliveryStartTime, setDeliveryStartTime] = React.useState<Dayjs | null>(null)
+  const [deliveryEndTime, setDeliveryEndTime] = React.useState<Dayjs | null>(null)
+
+  const goodTypes = [
+    { title: translate('newDelivery.goodType.pallets'), value: GoodType.pallets, quantity: 1 },
+    { title: translate('newDelivery.goodType.packages'), value: GoodType.packages, quantity: 2 },
+    { title: translate('newDelivery.goodType.pieces'), value: GoodType.pieces, quantity: 3 }
+  ]
+
+  const randomApprovedOn: string = '2024-07-04T10:06:12.594Z'
+  const randomCreatedOn: string = '2024-07-04T10:06:12.594Z'
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value)
+  }
+
+  const columnsData: Column[] = [
+    { key: 'number', title: translate('deliveries.newDeliveryTable.columnsData.number') },
+    { key: 'vendorName', title: translate('deliveries.newDeliveryTable.columnsData.vendorName') },
+    {
+      key: 'deliveryNumber',
+      title: translate('deliveries.newDeliveryTable.columnsData.deliveryNumber')
+    },
+    {
+      key: 'receptionNumber',
+      title: translate('deliveries.newDeliveryTable.columnsData.receptionNumber')
+    },
+    {
+      key: 'waitingGoods',
+      title: translate('deliveries.newDeliveryTable.columnsData.waitingGoods')
+    },
+    {
+      key: 'completedGoods',
+      title: translate('deliveries.newDeliveryTable.columnsData.completedGoods')
+    },
+    { key: 'markers', title: translate('deliveries.newDeliveryTable.columnsData.markers') },
+    { key: 'status', title: translate('deliveries.newDeliveryTable.columnsData.status') },
+    { key: 'approvedOn', title: translate('deliveries.newDeliveryTable.columnsData.approvedOn') },
+    { key: 'createdOn', title: translate('deliveries.newDeliveryTable.columnsData.createdOn') },
+    { key: 'actions', title: translate('vendors.table.actions'), minWidth: 50, align: 'right' }
+  ]
+
+  const rowData = [
+    {
+      number: 1,
+      vendorName: 'Bosch',
+      deliveryNumber: 1,
+      receptionNumber: 1,
+      waitingGoods: (
+        <Box sx={{ display: 'flex', alignItems: 'center' }}>
+          <Typography>1</Typography>
+          <InfoPopper content={<DeliveryGoodsInfo goodTypes={goodTypes} />} />
+        </Box>
+      ),
+      completedGoods: (
+        <Box sx={{ display: 'flex', alignItems: 'center' }}>
+          <Typography>1</Typography>
+          <InfoPopper content={<DeliveryGoodsInfo goodTypes={goodTypes} />} />
+        </Box>
+      ),
+      markers: <ChipsList items={markers} />,
+      status: <ChipsList items={['Изчакване']} color="default" />,
+      approvedOn: dayjs(randomApprovedOn).format('DD.MM.YYYY'),
+      createdOn: dayjs(randomCreatedOn).format('DD.MM.YYYY'),
```

Comment:
good now create an util called `dateHelpers` for example. In there have a function called `formatDate` which would except an argument of `string | Date` and a second one about the format where the default one will be `DD.MM.YYYY`.

And in there we will put functions that are related to dates but also use it hide `dayjs` as a fascade so if for example want to change to a new date formatter we would just change in this file only/

Reply:
Fixed